flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java

```diff
@@ -213,7 +234,16 @@ private RestClusterClient(
         this.waitStrategy = checkNotNull(waitStrategy);
         this.clusterId = checkNotNull(clusterId);
 
-        this.clientHAServices = checkNotNull(clientHAServices);
+        this.clientHAServices =
+                clientHAServicesFactory.create(
+                        configuration,
+                        new FatalErrorHandler() {
+                            @Override
+                            public void onFatalError(Throwable exception) {
+                                webMonitorLeaderRetriever.handleError(
+                                        new FlinkException(exception));
+                            }
```

Comment:
```suggestion
                        exception ->
                                webMonitorLeaderRetriever.handleError(
                                        new FlinkException(exception))
```

A bit less boilerplate.

Reply:
dose this means add more error msg?